_protected/app/system/modules/mail/views/base/tpl/main/msg.inc.tpl

```diff
@@ -14,19 +23,56 @@
   {{ $set_to = ($is_delete) ? 'setdelete' : 'settrash' }}
   {{ $label_txt = ($is_delete) ? t('Delete') : t('Trash') }}
 
-  <div class="center">
-    <dl>
-      <dt>{lang 'Author:'}</dt>
-      <dd>{{ $avatarDesign->get($usernameSender, $firstNameSender, null, 32) }}</dd>
-      <dt>{lang 'Subject:'}</dt>
-      <dd>{subject}</dd>
-      <dt>{lang 'Message:'}</dt>
-      <dd>{message}</dd>
-      <dt>{lang 'When:'}</dt>
-      <dd>{% Framework\Date\Various::textTimeStamp($msg->sendDate) %}</dd>
-    </dl>
+  {{ $subject = escape(str_replace('re ', '', $msg->title), true) }}
+
+  <div class="left">
+  <dl>
+    <dt class="msg-label">{lang 'From:'}</dt>
+    <dd>{{ $avatarDesign->get($usernameSender, $firstNameSender, null, 32) }}</dd>
+    <dd class="msg-from"><?php echo $usernameSender ?></dd>
+    <dt class="msg-label">{lang 'Subject:'}</dt>
+    <dd class="msg-cc">{subject}</dd>
+  </dl>
+    
+  {if !empty($msgsOld)}
+           
+  <h6>Previous Messages:</h6>
+
+  {each $msg in $msgsOld}
+  
+  {* Set Variables *}
+  {{ $usernameSender = (empty($msg->username)) ? PH7_ADMIN_USERNAME : escape($msg->username) }}
+  {{ $firstNameSender = (empty($msg->firstName)) ? PH7_ADMIN_USERNAME : escape($msg->firstName) }}
+  {{ $subject = escape(Framework\Security\Ban\Ban::filterWord($msg->title)) }}
+  {{ $message = Framework\Parse\Emoticon::init(Framework\Security\Ban\Ban::filterWord($msg->message)) }}
+  {{ $is_outbox = ($msg->sender == $member_id) }}
+  {{ $is_trash = (($msg->sender == $member_id && $msg->trash == 'sender') || ($msg->recipient == $member_id && $msg->trash == 'recipient') && !$is_outbox) }}
+  {{ $is_delete = ($is_outbox || $is_trash) }}
+  {{ $set_to = 'setdelete' }}
+  {{ $label_txt = t('Delete') }}
+  
+  {{ $subject = escape(str_replace('re ', '', $msg->title), true) }}
+    
+      <!-- Sent by User -->
+      <div class="msg-left">
+        <dl>
+          <dd class="msg-text-area">{message}</dd>
+           <div>{{ LinkCoreForm::display($label_txt, 'mail', 'main', $set_to, array('id'=>$msg->messageId)) }}
+          {if $is_trash} | {{ LinkCoreForm::display(t('Move to Inbox'), 'mail', 'main', 'setrestore', array('id'=>$msg->messageId)) }}{/if}</div>
+          {if $msg->sender === $member_id}
+          <dt class="msg-l-sent">Sent: &nbsp;{% Framework\Date\Various::textTimeStamp($msg->sendDate) %}</dt>
+          {else}
+          <dt class="msg-l-received">Received: &nbsp;{% Framework\Date\Various::textTimeStamp($msg->sendDate) %}</dt>
+          {/if}
+        </dl>
+        
+    {/each}
+    {/if}
+        <hr>
+       
```

Comment:
Since we display up to the 25 messages of the conversation before the 'reply' form, it might be a good idea to directly scroll down the page to the bottom (to the reply form). 
It's probably more "user friendly".

We can do it easily with jQuery like,
```js
    var sHash = location.hash.substr(1);
    var $target = $('html, body');
    $target.animate({scrollTop: $target.height()}, 1000);
```

Reply:
Or you could put the reply at the top and and sort ascending instead. I was thinking of using offset and pages of history but just seems unnecessary.

Reply:
That jQuery animate is a nice touch. 

And I have favicon, menu and inbox(2) all telling me how many unread messages I have.

Reply:
> Or you could put the reply at the top and and order by ASC instead. I was thinking of using offset and pages of history but just seems unnecessary.

If you move the reply form to the top, it might be harder for users to read the previous messages before replying. I think having the reply form at the bottom is still a good thing

Reply:
Yes I agree. I will find a way around the recipient and subject being added.